### PR TITLE
Deprecate pygame.image.tostring and fromstring

### LINKED ISSUE #2085: Deprecate `image.tostring`, `image.fromstring`
It's been a while since we introduced `tobytes` and `frombytes` as the better named equivalents of these functions, and I've noticed people in the wild actually using the new names, so it's probably time to deprecate the old ones:

- [ ] Make functions generate a deprecation warning when called
- [ ] Mark functions deprecated in the docs

### PR DESCRIPTION
**Pull request by SSS-Says-Snek**
Fixes #2085

Since `to/fromstring`'s gonna be deprecated, I also replaced usage of them in tests to `to/frombytes`
Finally, like I mentioned on the discord, because to/frombytes and to/fromstring's docs were the exact same and made it hard to follow, I replaced it with a redirection to `to/frombytes`

### CODE REVIEW

**Comment by Starbuck5** [comment]
Thanks for working on this!

My only review notes is for the docs:
- The functions with removed documentation should at least say something like "This is the same functionality as frombytes"
- The reference to the new function should be a sphinx link to the new function. For example, ":func:\`pygame.image.load()`" is a link to the function.

In case you haven't built the docs before, do `python setup.py docs` (with sphinx installed), and then access them with `python -m docs`

**Comment by SSS-Says-Snek** [comment]
Alright, I'll fix it when I get back

**Comment by SSS-Says-Snek** [comment]
Soo I won't be back for 6 more days💀

**Comment by novialriptide** [comment]
🚗

**Review by Starbuck5 — APPROVED** [review]
Looks good to me!

File: src_c/image.c [hunk 1]
```diff
@@ -1221,6 +1221,40 @@ image_frombytes(PyObject *self, PyObject *arg, PyObject *kwds)
...
+                     1) == -1) {
+        return NULL;
+    }
+
+    PyObject *result = image_frombytes(self, arg, kwarg);
+    if (result == NULL) {
+        return NULL;
+    }
+
+    return result;
```
Comment (ankith26) [inline src_c/image.c:1255]:
```suggestion
    return image_frombytes(self, arg, kwarg);
```
super minor nitpick. You can also do the same for the above function

**Review by ankith26 — APPROVED** [review]
LGTM, thanks for the PR! :tada: 

Left a minor nitpick-ey review, resolve at will
